pgpubsub/management/commands/listen.py

```diff
@@ -33,12 +59,25 @@ def handle(self, *args, **options):
         channel_names = options.get('channels')
         processes = options.get('processes') or 1
         recover = options.get('recover', False)
-        multiprocessing.set_start_method('fork', force=True)
-        connection.close()
-        for i in range(processes):
-            process = multiprocessing.Process(
-                name=f'pgpubsub_process_{i}',
-                target=listen,
-                args=(channel_names, recover),
+        worker = options.get('worker', False)
+        worker_start_method = options.get('worker_start_method')
+        autorestart_on_failure = not options.get('no_restart_on_failure')
+        if worker:
+            if processes > 1:
+                raise ValueError(
+                    f'Only 1 process is allowed with --worker option. Found {processes}'
+                )
+            listen(
+                channel_names,
+                recover,
+                autorestart_on_failure,
+                start_method=worker_start_method,
             )
-            process.start()
+        else:
+            for i in range(processes):
+                start_listen_in_a_process(
```

Comment:
A minor diff between this and the old logic is that with this version, we always close the connection (in `start_listen_in_a_process`) for each process in the loop, rather than once before the loop starts. Is that necessary?

Reply:
This will not change anything as the django connection maintains the state and `close` is idempotent and may be run multiple times - all further calls will be noop. This change has an advantage that when the method is used it is impossible to forget to close the connection. And we use it in multiple places now.